### PR TITLE
Implement a service for operating JobRole entity #462

### PR DESCRIPTION
**Pull request by jasonwong26**
## Changes

- Added `jobRole.service.js` file
- Added unit tests

## Notes

From the structure of the JobRole model, it looks like it is meant to be a single document rather than a collection.  For the `getAll()` method, I leveraged the `findeOne()` method to retrieve the single document.  Note that if there are multiple documents in a collection `findOne()` will return at random [link]https://mongoosejs.com/docs/api/model.html#model_Model.findOne).

I'm not familiar enough with Mongoose to say if there is a way to enforce the single document structure for `obRole`.  Not sure that we even need to spend time on this either if this entity is moving to a separate system.

## Questions

1. I noticed a number of EsLint warnings and errors.  Should I create a ticket to address these?
2. Similar question, but for package warnings?
3. I get an unhandled error message when I run the `test:all` command from the root directory.  Does anyone else get this?  Or is there a setup step I missed?  (I'm guessing this is coming from the final part of the command: `np run test:cy`

```
[nodemon] 2.0.4
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node server.js`
Connected to Slack
(node:2348) UnhandledPromiseRejectionWarning: Error: An API error occurred: account_inactive
    at Object.platformErrorFromResult (D:\cyber\Documents\Programming\Github\VRMS\backend\node_modules\@slack\web-api\dist\errors.js:51:33)
    at WebClient.apiCall (D:\cyber\Documents\Programming\Github\VRMS\backend\node_modules\@slack\web-api\dist\WebClient.js:152:28)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:2348) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:2348) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Mongoose connected from runServer() and is listening on 4000
REQUEST SHOULD CONTAIN CUSTOM HEADER BUT IT ISN'T FOUND
HEAD / 401 2.494 ms - 12
REQUEST SHOULD CONTAIN CUSTOM HEADER BUT IT ISN'T FOUND
HEAD / 401 0.429 ms - 12
```

### CODE REVIEW

**Comment by akibrhast** [comment]
I get an unhandled error message when I run the test:all command from the root directory. Does anyone else get this? Or is there a setup step I missed?
Yes. No.

**Review by drubgrubby — COMMENTED** [review]
@jasonwong26

I'm guessing that this service also needs to be added to `backend/services/index.js`

**Comment by jasonwong26** [comment]
Hi @drubgrubby ,

Pushed the changes we discussed, please take another look.  Appreciate you catching it!

File: backend/globalConfig.json [hunk 1]
```diff
@@ -1 +1 @@
-{"mongoUri":"mongodb://127.0.0.1:3170/jest?","mongoDBName":"jest"}
+{"mongoUri":"mongodb://127.0.0.1:21407/jest?","mongoDBName":"jest"}
```
Comment (drubgrubby) [inline backend/globalConfig.json]: I don't think this globalConfig.json needs to be part of this PR.

**Review by drubgrubby — CHANGES_REQUESTED** [review]
Thanks for adding the service to the index.js.

**Review by drubgrubby — APPROVED** [review]
Looks good.

**Pull request merged**